cli/src/xmltools.ts

```diff
@@ -0,0 +1,121 @@
+import * as xmldom from "xmldom";
+import { evaluate, mapXpathResult, flatmapXpathResult } from "./xpathutils";
+const xpath = require("xpath");
+
+/**
+ * createInterface parses an xml string and generates code for a TypeScript interface.
+ */
+export function createInterface(interfaceName: string, xml: string): string {
+  return interfacefmt(parseXML(interfaceName, xml));
+}
+
+export interface GeneratedInterface {
+  name: string;
+  fields: GeneratedField[];
+}
+
+export interface GeneratedField {
+  name: string;
+  type: string;
+  comment?: string;
+  optional: boolean;
+
+  subfields?: GeneratedField[];
+}
+
+function interfacefmt(iface: GeneratedInterface): string {
```

Comment:
It is unclear to me what **fmt** means in `interfacefmt` and `fieldfmt`.

There is a best practice that you should not use abbreviations in naming, because it can cause confusion. Especially if someone takes over the project later.

Reply:
I see `commentfmt` too.